It appears that there is some discrepancy with `test262` in particular with this patch. I haven't dug deeply into it, but here are the numbers:

Without the patch:
```
===
=== 1107 tests failed
===
>>> 57187 base tests produced 93466 (163%) non-filtered tests
>>> 93466 tests ran
```

With the patch:
```
===
=== 1754 tests failed
===
>>> 57187 base tests produced 93466 (163%) non-filtered tests
>>> 93466 tests ran
```

An example of a failing test with the patch can be seen:
```
${OUT}/d8 test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js

test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:41: Error: broken promise
    throw new Error('broken promise');
    ^
Error: broken promise
    at Promise.get (test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:41:11)
    at g.return (<anonymous>)
    at test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:45:10
```

and without the patch:
```
${OUT}/d8 test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js

test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:54: ReferenceError: $DONE is not defined
  .then($DONE, $DONE);
        ^
ReferenceError: $DONE is not defined
    at test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:54:9

test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:51: ReferenceError: assert is not defined
      assert.sameValue(err.message, 'broken promise');
      ^
ReferenceError: assert is not defined
    at test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:51:7

1 pending unhandled Promise rejection(s) detected.
```

It may be that without this patch, the test worked accidentally. However, the output of this particular test in the latest commit in upstream V8 as of today seems to match, so I'm inclined to believe that something else is going on:
```
test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:54: ReferenceError: $DONE is not defined
  .then($DONE, $DONE);
        ^
ReferenceError: $DONE is not defined
    at test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:54:9

test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:51: ReferenceError: assert is not defined
      assert.sameValue(err.message, 'broken promise');
      ^
ReferenceError: assert is not defined
    at test/test262/data/test/built-ins/AsyncGeneratorPrototype/return/return-suspendedStart-broken-promise.js:51:7

1 pending unhandled Promise rejection(s) detected.
```

--- a/src/objects/code-inl.h
+++ b/src/objects/code-inl.h
@@ -171,7 +171,14 @@ Address Code::InstructionEnd(Isolate* isolate, Address pc) const {
 }
 
 int Code::GetOffsetFromInstructionStart(Isolate* isolate, Address pc) const {
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(__aarch64__)
+  // NOTE(zyj20): Take into account that the LSB of sealed PCC is set.
+  // The resulting capability could be invalid but it won't be
+  // used as a pointer anyways.
+  const Address offset = (pc & ~1) - InstructionStart(isolate, pc);
+#else
   const Address offset = pc - InstructionStart(isolate, pc);
+#endif
   DCHECK_LE(offset, instruction_size());
   return static_cast<int>(offset);
 }

